src/main/java/io/cdap/plugin/gcp/gcs/GCSPath.java

```diff
@@ -86,7 +87,7 @@ public int hashCode() {
    * @throws IllegalArgumentException if the path string is invalid
    */
   public static GCSPath from(String path) {
-    if (path.isEmpty()) {
+    if (Strings.isNullOrEmpty(path)) {
```

Comment:
why is this change needed? normally we assume the input is non-null unless it's annotated as nullable.

Reply:
If it is null, as it was before this bug fix for GCSSource, then the NullPointerException thrown is harder to debug than an IllegalArgumentException.